### ISSUE TITLE
User tags are disabled if followed by comma, colon, ...

### ISSUE THREAD

**Issue by HenrikBengtsson** [go-gitea/gitea#8465]
- Gitea version (or commit ref): 1.9.4
- Git version: N/A
- Operating system: Linux
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (<https://try.gitea.io/HenrikBengtsson/test/issues/17>)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

```
Putting a comma, colon, or a semicolon after a user tag causes the tag to _not_ link to that user, e.g.

1. @hb works, and
2. so does @hb.
3. However, using @hb, @hb:, or @hb; do _not_ work.

Similarly, putting a user tag in curly brackets disables the link, e.g.

4. Moreover, (@hb) and [@hb] work, but
4. {@hb} doesn't.
```

As a reference, all of the above works on GitHub.

(I fenced the description to make it easy to cut'n'paste for testing).

## Screenshots

![gitea_1 9 4_tagging_bug](https://user-images.githubusercontent.com/1616850/66672668-af3c2900-ec13-11e9-9ffd-a7876cd85462.png)

**Comment by guillep2k** [comment]
This could be fixed in the regular expression that parses mentions. I'm not sure it's always advisable, though. I didn't analyze this myself, but perhaps there are many cases where text looks like a mention but it's not. As a mention immediately sends an e-mail, false positives could be very annoying.

Probably a good trade-off would be to accept `,`, `;`, `:` as long as they are followed by a spacing character (i.e. space, tab, end of line).

**Comment by HenrikBengtsson** [comment]
A false *negative* that I keep hitting on a daily basis is when I mention someone as:

> Hi @hb, can you please have a look at this?

Much less common, we have at times wanted to do:

* @alice: do this...
* @bob: something else...

...

> Probably a good trade-off would be to accept `,`, `;`, `:` as long as they are followed by a spacing character (i.e. space, tab, end of line).

That sounds like a reasonable tradeoff.

**Comment by guillep2k** [comment]
This can be closed as per #8889
